express-api/src/controllers/projects/projectsController.ts

```diff
@@ -22,7 +25,37 @@ export const getDisposalProject = async (req: Request, res: Response) => {
    *   "bearerAuth" : []
    * }]
    */
-  return stubResponse(res);
+  const projectId = Number(req.params.projectId);
+  if (isNaN(projectId)) {
+    return res.status(400).send('Project ID was invalid.');
+  }
+  const project = await projectServices.getProjectById(projectId);
+  if (!project) {
+    return res.status(404).send('Project matching this internal ID not found.');
+  }
+  const parcelIds = project.ProjectProperties?.filter((p) => p.ParcelId != null).map(
+    (p) => p.ParcelId,
+  );
+  const buildingIds = project.ProjectProperties?.filter((b) => b.BuildingId != null).map(
+    (b) => b.BuildingId,
+  );
+  const parcels = await AppDataSource.getRepository(Parcel).find({
+    relations: {
+      Agency: true,
+      Evaluations: true,
+      Fiscals: true,
+    },
+    where: { Id: In(parcelIds ?? []) },
+  });
+  const buildings = await AppDataSource.getRepository(Building).find({
+    relations: {
+      Agency: true,
+      Evaluations: true,
+      Fiscals: true,
+    },
+    where: { Id: In(buildingIds ?? []) },
+  });
```

Comment:
I think you could also do this (albeit with the parcels and buildings together) by querying ProjectProperties:
``` ts
  const result = await AppDataSource.getRepository(ProjectProperty).find({
    where: { ProjectId: project.Id },
    relations: {
      Parcel: {
        Evaluations: true,
        Fiscals: true,
        Agency: true,
      },
      Building: {
        Evaluations: true,
        Fiscals: true,
        Agency: true,
      },
    },
  });
```

Reply:
Or like this from the project originally if we want to update the getProject service:
``` ts
const result = await AppDataSource.getRepository(Project).find({
    where: { Id: project.Id },
    relations: {
      ProjectProperties: {
        Parcel: {
          Agency: true,
          Evaluations: true,
          Fiscals: true
        },
        Building: {
          Agency: true,
          Evaluations: true,
          Fiscals: true
        }
      }
    },
  });
```

Reply:
The second example is similar to what Sharala had previously and it was not working. First example is good though, does simplify the logic a bit. I'll give that a try.

Reply:
> The second example is similar to what Sharala had previously and it was not working. First example is good though, does simplify the logic a bit. I'll give that a try.

Not working how? I seem to be able to get the project object back with all the relations.

Reply:
Updated to that first method, you're right it works fine